src/main/java/no/nav/folketrygdloven/beregningsgrunnlag/fordel/OmfordelBGForArbeidsforhold.java

```diff
@@ -17,50 +20,91 @@
+                resterendeBeløpÅFlytte = flyttBGFraAktivitetTilArbeid(aktivitet, resterendeBeløpÅFlytte, arbeidMedFlyttbartBeløp, flyttbartBeløp);
             } else {
-                beløpSomSkalOmfordelesTilAktivitet = omfordelDelerAvGrunnlagetFraArbeidsforhold(aktivitet, beløpSomSkalOmfordelesTilAktivitet, arbeidMedOmfordelbartBg, bgForArbeid);
+                resterendeBeløpÅFlytte = flyttBGFraAktivitetTilArbeid(aktivitet, resterendeBeløpÅFlytte, arbeidMedFlyttbartBeløp, resterendeBeløpÅFlytte);
             }
-            resultater.put("fordeltPrÅr", arbeidMedOmfordelbartBg.getGradertFordeltPrÅr());
-            resultater.put("arbeidsforhold", arbeidMedOmfordelbartBg.getBeskrivelse());
-            arbeidsforholdMedOmfordelbartBGOpt = finnArbeidsforhold.finn(beregningsgrunnlagPeriode);
+            resultater.put("fordeltPrÅr", arbeidMedFlyttbartBeløp.getFordeltPrÅr());
+            resultater.put("arbeidsforhold", arbeidMedFlyttbartBeløp.getBeskrivelse());
+            arbforholdMedFlyttbartBeløpOpt = finnArbeidsforhold.finn(beregningsgrunnlagPeriode);
         }
         return resultater;
     }
 
-    private static BigDecimal omfordelDelerAvGrunnlagetFraArbeidsforhold(BeregningsgrunnlagPrArbeidsforhold aktivitet, BigDecimal beløpSomSkalOmfordeles, BeregningsgrunnlagPrArbeidsforhold arbeidMedOmfordeltbartBg, BigDecimal bgForArbeid) {
-        BigDecimal fordelt = bgForArbeid.subtract(beløpSomSkalOmfordeles);
-        BeregningsgrunnlagPrArbeidsforhold.builder(arbeidMedOmfordeltbartBg)
-            .medFordeltPrÅr(fordelt);
-        omfordelBGTilAktivitet(aktivitet, beløpSomSkalOmfordeles);
-        beløpSomSkalOmfordeles = BigDecimal.ZERO;
-        return beløpSomSkalOmfordeles;
+    private BigDecimal finnFlyttbartBeløp(BeregningsgrunnlagPrArbeidsforhold arbeidMedOmfordelbartBg) {
+        BigDecimal bgForArbeidFratrektNaturalytelse = finnBgFratrektTilkommetNaturalytelse(arbeidMedOmfordelbartBg);
+        BigDecimal refusjonskrav = arbeidMedOmfordelbartBg.getRefusjonskravPrÅr().orElse(BigDecimal.ZERO);
+        return bgForArbeidFratrektNaturalytelse.subtract(refusjonskrav);
     }
 
-    private static BigDecimal omfordelHeleGrunnlagetFraArbeidsforhold(BeregningsgrunnlagPrArbeidsforhold aktivitet, BigDecimal beløpSomSkalOmfordeles, BeregningsgrunnlagPrArbeidsforhold arbeidMedOmfordelbartBg, BigDecimal refusjonskrav, BigDecimal omfordelbartBeløp) {
-        BeregningsgrunnlagPrArbeidsforhold.builder(arbeidMedOmfordelbartBg)
-            .medFordeltPrÅr(refusjonskrav);
-        omfordelBGTilAktivitet(aktivitet, omfordelbartBeløp);
-        beløpSomSkalOmfordeles = beløpSomSkalOmfordeles.subtract(omfordelbartBeløp);
-        return beløpSomSkalOmfordeles;
+    private BigDecimal finnBgFratrektTilkommetNaturalytelse(BeregningsgrunnlagPrArbeidsforhold arbeidMedOmfordelbartBg) {
+        return arbeidMedOmfordelbartBg.getBruttoPrÅr().orElse(BigDecimal.ZERO)
```

Comment:
Vi trenger ikke legge til bortfalt nat.ytelse her altså?

Reply:
Nei, for problemet er at vi ikkje har lyst til å endre på naturalytelse (for å ikkje gjere logikken alt for kompleks og fortsatt kunne dekke omsorgspenger foreløpig). Då er vi avhengig av å ikkje flytte grunnlaget som svarer til tilkommet naturalytelse fordi dette blir trekt i fra grunnlaget (og man vil ende opp med eit negativt beregningsgrunnlag). Bortfalt treng vi ikkje bry oss om, for dette blir berre lagt til.